src/history.rs

```diff
@@ -48,56 +49,65 @@ const_assert_eq!(4, align_of::<ModalityProbeInstant>());
 
 #[cfg(target_pointer_width = "32")]
 const_assert_eq!(
-    size_of::<ProbeId>()
+    size_of::<u32>()
+        + size_of::<ProbeId>()
+        + size_of::<LogBuffer<'_>>()
         + size_of::<u32>()
-        + size_of::<FixedSliceVec<'_, LogicalClock>>()
         + size_of::<LogicalClock>()
-        + size_of::<LogEntryRingBuffer<'_>>()
-        + size_of::<u32>()
-        + size_of::<u64>()
-        + size_of::<RestartSequenceCounter<'_>>(),
+        + size_of::<FixedSliceVec<'_, LogicalClock>>()
+        + size_of::<RestartSequenceCounter<'_>>()
+        + size_of::<u64>(),
     size_of::<DynamicHistory>()
 );
 
-// On 64-bit, size is rounded up to the nearest multiple of its alignment:
-// size of the fields is 116 bytes, which rounds up to
-// the next 8-byte alignment at 120 bytes (a difference of 4 bytes)
-#[cfg(not(target_pointer_width = "32"))]
+// 4 bytes of padding required before clocks list to get FixedSliceVec to 8 byte align
+#[cfg(target_pointer_width = "64")]
 const_assert_eq!(
-    4 + size_of::<ProbeId>()
+    size_of::<u32>()
+        + size_of::<ProbeId>()
+        + size_of::<LogBuffer<'_>>()
         + size_of::<u32>()
-        + size_of::<FixedSliceVec<'_, LogicalClock>>()
         + size_of::<LogicalClock>()
-        + size_of::<LogEntryRingBuffer<'_>>()
-        + size_of::<u32>()
-        + size_of::<u64>()
-        + size_of::<RestartSequenceCounter<'_>>(),
+        + 4
+        + size_of::<FixedSliceVec<'_, LogicalClock>>()
+        + size_of::<RestartSequenceCounter<'_>>()
+        + size_of::<u64>(),
     size_of::<DynamicHistory>()
 );
 
 /// Manages the core of a probe in-memory implementation
 /// backed by runtime-sized arrays of current logical clocks
 /// and probe log items
+///
+/// Note: overwrite_priority, probe_id, and log must be accessed by the debug collector using direct memory access,
+/// so they must be public fields (in order for the debug collector to calculate their offsets).
```

Comment:
Rather than making these fields public, could we instead expose the offsets through some feature-flag-gated functions in modality-probe?  Seems like that would be less prone to the sundry issues that come with exposing struct internals completely.